File: leetcode/day3/python/so.py
from itertools import product
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for index, num in enumerate(nums):
            if ((target - num) in m):
                return [m[target - num], index]
            else:
                m[num] = index
        return [0, 0]

File: leetcode/day3/cpp/so.cpp
#include "so.h"
#include <map>

vector<int> Solution::twoSum(vector<int> &nums, int target)
{

	map<int, int> m = {};
	int index = 0;
	vector<int> r;

	for (int &num : nums)
	{
		auto pos = m.find(target - num);

		if (pos == m.end())
		{
			m[num] = index;
		}
		else
		{
			r = {pos->second, index};
		}
		index++;
	}
	return r;
}